common/app/model/Badges.scala

```diff
@@ -3,10 +3,40 @@ package model
 import conf.switches.Switches
 import conf.{Configuration, Static}
 import layout.FaciaContainer
+import java.security.MessageDigest
+import java.math.BigInteger
 
-case class Badge(seriesTag: String, imageUrl: String, classModifier: Option[String] = None)
+
+trait BaseBadge{
+  def imageUrl(id: String): String
+  def classModifier: Option[String]
+  def isBadgeTag(tag: String):Boolean
+}
+
+case class Badge(seriesTag: String, staticImageUrl: String, classModifier: Option[String] = None) extends BaseBadge {
+  def isBadgeTag(tag: String) = seriesTag == tag
+  def imageUrl(id: String) = staticImageUrl
```

Comment:
That's my point. Signature conveys the idea the url depends on this string which is not always the case